core/src/test/scala/org/apache/spark/MapOutputTrackerSuite.scala

```diff
@@ -29,7 +29,11 @@ import org.apache.spark.shuffle.FetchFailedException
 import org.apache.spark.storage.{BlockManagerId, ShuffleBlockId}
 
 class MapOutputTrackerSuite extends SparkFunSuite {
+
   private val conf = new SparkConf
+  val env = mock(classOf[SparkEnv])
```

Comment:
It's a little weird that we have to modify this. I kind of understand why we don't want to thread SparkEnv or another setting into the `MapStatus` constructor but maybe we could avoid the confusing NPEs that this prevents by doing something like

```
Option(SparkEnv.get).map(_.conf.get(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD)).getOrElse(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD.default)
```

thereby minimizing changes to other files / tests.